sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -958,17 +958,115 @@ this extension.
-`dir` is a separate element in the `build_options` list.
-|===
+kernels using external libraries.
+
+===== `--define-macro=<name>[=<value>]` (`-D<name>[=<value>]`)
+
+Define macro `<name>`, optionally to the given `<value>`.
+
+===== `--undefine-macro=<name>` (`-U<name>`)
+
+Undefine macro `<name>`.
+
+==== Diagnostic options
+
+The usual warning (starting with `-W`) and remark (starting with `-R`) emission
+options are supported. Note that the `save_log` property must be used to obtain
+detailed output from the compilation process.
+
+==== SYCL-specific options
+
+===== `-Xs<arg>`
+
+Pass `<arg>` to the target compiler. When using `-Xs<arg>`, a `-` is prepended
+to `<arg>` before handing it to the target compiler. Otherwise, `<arg>` is
+passed on unmodified.
+
+For example, the following forms are equivalent:
+
+[source,c++]
+----
+build_options{{
+  {"-XsDFOO=bar"},
+  {"-Xs"}, {"-DFOO=bar"}
+}};
+----
+
+===== `-fsycl-rtc-mode`
+
+Relax the requirement that parameter types for free-function kernels must be
+forward-declarable.
+
+=== Unsupported `build_options` when the language is `sycl`
+
+The SYCL runtime compiler currently does not support the following {dpcpp}
+options and associated features.
+
+==== Ahead-of-time compilation
+
+With the exception of the `-Xs` option, all other options to control
+ahead-of-time (AOT) compilation, such as `-fsycl-targets`, are unsupported.
+
+[_Note:_ 
+The kernels in a SYCL source string are compiled automatically to native code
+for all devices passed to the `build` function (see section "New free functions
+to create and build kernel bundles").
+_{endnote}_]
+
+==== Other compiler actions
+
+As the {dpcpp} frontend is integrated tightly in the runtime compilation
+pipeline, all options to change the compiler action (e.g. `-c`, `-S`) or output
+files (`-o`) are unsupported. Options related to linking (e.g. `-L`) are
+unsupported, including the SYCL-specific `-fsycl-link` action.
+
+==== Debug info
+
+Currently, debugging of runtime-compiled kernels is not supported. Hence, all
+options related to the generation and embedding of debug information (e.g. `-g`)
+are unsupported.
+
+==== `invoke_simd`
+
+While the SYCL runtime compiler supports ESIMD kernels and source strings
+containing a mix of SYCL and ESIMD kernels, the `invoke_simd` functionality is
+currently unsupported. The related `-fno-sycl-device-code-split-esimd` option is
+unsupported.
+
+==== Sanitizers
+
+Device, memory and thread sanitizers are unsupported for runtime-compiled
+kernels. The `-fsanitize=` option is unsupported.
+
+=== Caching
+
+The `kernel_compiler` implementation in {dpcpp} supports persistent caching. To
+enable it, set the the environment variable `SYCL_CACHE_PERSISTENT=1`. The
+location of the cache can be changed by setting `SYCL_CACHE_DIR`. Refer to
+https://intel.github.io/llvm/design/KernelProgramCache.html#persistent-cache for
+more details on how to control the cache.
```

Comment:
Has this been reviewed somewhere else, perhaps as part of the design of the run-time compiler?

I'm a little surprised that JIT compilation of kernels embedded in the binary would use the same cache as explicit run-time compilation of kernels.  I would have expected separate controls.

Reply:
The implementation PR is https://github.com/intel/llvm/pull/16823 and has been approved by @cperkinsintel.

Reply:
Okay, thanks for confirming that.